File: backend/app/routers/wishlist.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import List
from ..database import get_db
from .. import schemas, crud

router = APIRouter(prefix="/wishlist", tags=["Wishlist"])

@router.post("/", response_model=schemas.WishlistResponse)
def add_to_wishlist(wishlist: schemas.WishlistAdd, db: Session = Depends(get_db)):
    """Add a painting to user's wishlist"""
    try:
        crud.add_to_wishlist(db, wishlist.user_id, wishlist.painting_id)
        return schemas.WishlistResponse(
            success=True,
            message="Painting added to wishlist"
        )
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.get("/", response_model=List[schemas.PaintingResponse])
def get_wishlist(user_id: str = Query(..., description="User ID"), db: Session = Depends(get_db)):
    """Get all paintings in user's wishlist"""
    try:
        paintings = crud.get_user_wishlist(db, user_id)
        return paintings
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.delete("/", response_model=schemas.WishlistResponse)
def remove_from_wishlist(wishlist: schemas.WishlistAdd, db: Session = Depends(get_db)):
    """Remove a painting from user's wishlist"""
    try:
        removed = crud.remove_from_wishlist(db, wishlist.user_id, wishlist.painting_id)
        if not removed:
            raise HTTPException(status_code=404, detail="Painting not found in wishlist")
        return schemas.WishlistResponse(
            success=True,
            message="Painting removed from wishlist"
        )
    except HTTPException:
        raise
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

File: frontend/src/api/wishlist.ts
import { apiRequest, API_ENDPOINTS } from './config';
import { Painting } from './recommend';

// Types matching backend schemas
export interface WishlistAddRequest {
  user_id: string;
  painting_id: string;
}

export interface WishlistResponse {
  success: boolean;
  message: string;
}

// Wishlist Service
export const wishlistService = {
  /**
   * Add a painting to user's wishlist
   * POST /wishlist/
   */
  async addToWishlist(userId: string, paintingId: string): Promise<WishlistResponse> {
    return apiRequest<WishlistResponse>(API_ENDPOINTS.WISHLIST, {
      method: 'POST',
      body: JSON.stringify({
        user_id: userId,
        painting_id: paintingId,
      }),
    });
  },

  /**
   * Get all paintings in user's wishlist
   * GET /wishlist/?user_id={user_id}
   */
  async getWishlist(userId: string): Promise<Painting[]> {
    return apiRequest<Painting[]>(`${API_ENDPOINTS.WISHLIST}?user_id=${userId}`);
  },

  /**
   * Remove a painting from user's wishlist
   * DELETE /wishlist/?user_id={user_id}&painting_id={painting_id}
   */
  async removeFromWishlist(userId: string, paintingId: string): Promise<WishlistResponse> {
    return apiRequest<WishlistResponse>(
      `${API_ENDPOINTS.WISHLIST}?user_id=${userId}&painting_id=${paintingId}`,
      { method: 'DELETE' }
    );
  },
};
